crates/common/fork_choice/src/store.rs

```diff
@@ -428,22 +532,24 @@ impl Store {
 
         // Attestations can only affect the fork choice of subsequent slots.
         // Delay consideration in the fork choice until their slot is in the past.
-        ensure!(self.get_current_slot() >= attestation.data.slot + 1);
+        ensure!(self.get_current_slot()? >= attestation.data.slot + 1);
 
         Ok(())
     }
 
     pub fn store_target_checkpoint_state(&mut self, target: Checkpoint) -> anyhow::Result<()> {
-        // Store target checkpoint state if not yet seen
-        if let Entry::Vacant(entry) = self.checkpoint_states.entry(target) {
-            let mut base_state = self.block_states[&target.root].clone();
-            let target_slot = compute_start_slot_at_epoch(target.epoch);
-            if base_state.slot < target_slot {
-                base_state.process_slots(target_slot)?;
+        if self.db.checkpoint_states_provider().get(target)?.is_none() {
+            if let Some(base_state) = self.db.beacon_state_provider().get(target.root)? {
+                let mut base_state = base_state;
+                let target_slot = compute_start_slot_at_epoch(target.epoch);
+                if base_state.slot < target_slot {
+                    base_state.process_slots(target_slot)?;
+                }
+                self.db
+                    .checkpoint_states_provider()
+                    .insert(target, base_state)?;
```

Comment:
```suggestion
        if self.db.checkpoint_states_provider().get(target)?.is_some() {
            return Ok(());
        }
        
        let Some(mut base_state) = self.db.beacon_state_provider().get(target.root)? else {
            return Ok(());
        }
        
        let target_slot = compute_start_slot_at_epoch(target.epoch);
        if base_state.slot < target_slot {
            base_state.process_slots(target_slot)?;
        }
        self.db
            .checkpoint_states_provider()
            .insert(target, base_state)?;
```

We can make this code a little less indented